src/ConnectionFromMongoCursor.js

```diff
@@ -170,7 +170,10 @@ const connectionFromMongoCursor = async ({
   cursor.skip(skip);
   cursor.limit(limit);
 
-  const slice = await cursor.exec();
+  //avoid large object retrieval from database
+  const clonedCursor = cursor.find().merge(cursor).select({ _id: 1 })
```

Comment:
do we need to use `merge` ?

can't this be

`cursor.find().select({_id: 1})`?